### PR TITLE
Update all unit test packages, set node engine rules, new circleci workflow, pin to stable

### PR DESCRIPTION
**Pull request by unlikelyzero**
Resolves https://github.com/nasa/openmct/issues/3952
Resolves https://github.com/nasa/openmct/issues/3860

Will enforce https://github.com/nasa/openmct/issues/3966

## In this PR:

- [x] Add matrix build to test all supported versions of node
- [x] Add firefox and chrome support to karma
- [x] Update all karma and jasmine related dependencies
- [x] Pin to stable ubuntu
- [x] Pin to stable Chrome
- [x] Pin to stable versions of node
- [x] Restrict node versions https://github.com/nasa/openmct/issues/3952
- [x] Automatically run the latest version of chrome by default https://github.com/nasa/openmct/issues/3860
- [x] Add junit reporter to karma
- [x] Update Github Action to use supported version of node
- [x] Add test result storage to CircleCI for CircleCI Test Insights functionality
- [ ] Determine how to cache browsers to save download step time
- [x] Get Chrome tests to pass as consistently as Firefox
- [x] ~Add Chrome Canary to Matrix~. Moved to separate task https://github.com/nasa/openmct/issues/3967

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?

### Author Checklist

* [x] Changes address original issue?
* [x] Unit tests included and/or updated with changes?
* [x] Command line build passes?
* [ ] Has this been smoke tested?
* [ ] Testing instructions included in associated issue?

### CODE REVIEW

File: karma.conf.js [hunk 1]
```diff
@@ -59,14 +59,19 @@ module.exports = (config) => {
...
         },
         customLaunchers: {
             ChromeDebugging: {
                 base: 'Chrome',
                 flags: ['--remote-debugging-port=9222'],
                 debug: true
+            },
+            FirefoxESR: {
+                base: 'FirefoxHeadless',
+                name: 'FirefoxESR'
```
Comment (unlikelyzero) [inline karma.conf.js:74]: This will allow us to allow us to run Firefox ESR and Firefox latest locally in the future

File: .circleci/config.yml [hunk 2]
```diff
@@ -1,36 +1,73 @@
...
+          steps:
+            - browser-tools/install-chrome:
+                replace-existing: false         
+      - save_cache:
+          key: deps-{{ .Branch }}-{{ checksum "package.json" }}
+          paths:
+            - ~/.npm
+            - ~/.cache
+            - node_modules
+      - run: npm run test:coverage -- --browsers=<<parameters.browser>> || <<parameters.always-pass>>
```
Comment (unlikelyzero) [inline .circleci/config.yml:45]: || << parameters.always-pass >> will run the previous statement with OR logic. In effect, we can run as many or as few of our supported build matrix but chose to fail on specific builds

Comment (unlikelyzero) [reply .circleci/config.yml:45]: We could also change this parameter to `ignore-result` if that would be clearer

File: .circleci/config.yml [hunk 3]
```diff
@@ -1,36 +1,69 @@
...
-        - run:
-            name: npm run lint
-            command: npm run lint
-        - store_artifacts:
-            path: dist
-            prefix: dist
-
+      - checkout
+      - restore_cache:
+          key: deps-{{ .Branch }}--<< parameters.node-version >>--{{ checksum "package.json" }}
```
Comment (unlikelyzero) [inline .circleci/config.yml:22]: this allows us to cache by node-version and branch without a package-lock.json file

File: karma.conf.js [hunk 4]
```diff
@@ -23,9 +23,9 @@
 /*global module,process*/
 
 const devMode = process.env.NODE_ENV !== 'production';
-const browsers = [process.env.NODE_ENV === 'debug' ? 'ChromeDebugging' : 'FirefoxHeadless'];
+const browsers = [process.env.NODE_ENV === 'debug' ? 'ChromeDebugging' : 'ChromeHeadless'];
 const coverageEnabled = process.env.COVERAGE === 'true';
-const reporters = ['progress', 'html'];
+const reporters = ['progress', 'html', 'junit'];
```
Comment (unlikelyzero) [inline karma.conf.js:28]: necessary for circleci test results

**Review by akhenry — APPROVED** [review]
### Reviewer Checklist

1. Changes appear to address issue? Y
2. Appropriate unit tests included? N/A - Work is on the test toolchain itself
3. Code style and in-line documentation are appropriate? Y
4. Commit messages meet standards? Y
5. Has associated issue been labelled `unverified`? (only applicable if this PR closes the issue)

**Pull request merged**